mobile/src/main/java/ch/epfl/sdp/mobile/application/chess/Color.kt

```diff
@@ -1,11 +1,53 @@
 package ch.epfl.sdp.mobile.application.chess
 
-/** The classical colors of a game of chess. */
-enum class Color {
+/**
+ * The classical colors of a game of chess.
+ *
+ * @param normalizeDelta a function which normalizes the [Delta].
+ * @param normalizePosition a function which normalizes the [Position].
+ * @param opposite a function that returns the opposite color.
+ */
+enum class Color(
+    private val normalizeDelta: Delta.() -> Delta,
+    private val normalizePosition: Position.() -> Position,
+    private val opposite: () -> Color,
+) {
 
   /** The black color. */
-  Black,
+  Black(
+      normalizeDelta = { Delta(x = x, y = -y) },
```

Comment:
The `x` and `-y` come from the `Delta.()` receiver (since `normalizeDelta` is a `Delta.() -> Delta`. When normalised, a `Delta` for the black player should have a vertical symmetry applied, hence why the `y` coordinate is inverted.